srtcore/api.cpp

```diff
@@ -263,9 +357,10 @@ SRTSOCKET CUDTUnited::newSocket()
       throw CUDTException(MJ_SYSTEMRES, MN_MEMORY, 0);
    }
 
-   enterCS(m_IDLock);
-   ns->m_SocketID = -- m_SocketIDGenerator;
-   leaveCS(m_IDLock);
+    {
+        CGuard guard(m_IDLock);
+        ns->m_SocketID = generateSocketID();
```

Comment:
What if this CGuard moves inside the generateSocketID() function? 
`ns->m_SocketID` will have to be passed inside the function then?

Reply:
Good idea. The `ns->m_SocketID` is not a problem - this is a newly created object and this function is at the moment the only one that can access it.